androidplot-core/src/main/java/com/androidplot/Plot.java

```diff
@@ -433,73 +433,54 @@ private void loadAttrs(AttributeSet attrs, int defStyle) {
 
             Field styleableFieldInR = null;
             TypedArray typedAttrs = null;
-
-            final String appPkg = getContext().getPackageName();
-            Class styleableClass = null;
+            
+            Class styleableClass = R.styleable.class;
+            String styleableName = getClass().getName().substring(BASE_PACKAGE.length());
+            styleableName = styleableName.replace('.', '_');
             try {
                 /**
-                 * Need to retrieve R$styleable.class dynamically to avoid exceptions
-                 * in environments where this class does not exist, such as .jar users
-                 * that have not merged the library's attrs.xml with their own.
+                 * Use reflection to safely check for the existence of styleable defs for Plot
+                 * and it's derivatives.  This safety check is necessary to avoid runtime exceptions
+                 * in apps that don't include Androidplot as a .aar and won't have access to
+                 * the resources defined in the core library.

```

Comment:
hmm, maybe it would be an option to default to `R.styleable.class` and add the fallback by fetching from the package name, but as mentioned, the package name isn't the correct identifier for the resource class, although it will work in the most cases